pruner/params.go

```diff
@@ -0,0 +1,41 @@
+package pruner
+
+import (
+	"time"
+)
+
+type Option func(*Params)
+
+type Params struct {
+	// gcCycle is the frequency at which the pruning Service
+	// runs the ticker. If set to 0, the Service will not run.
+	gcCycle time.Duration
+}
+
+func DefaultParams() Params {
+	return Params{
+		gcCycle: time.Minute * 5,
+	}
+}
+
+// WithGCCycle configures how often the pruning Service
+// triggers a pruning cycle.
+func WithGCCycle(cycle time.Duration) Option {
+	return func(p *Params) {
+		p.gcCycle = cycle
+	}
+}
+
+// WithDisabledGC disables the pruning Service's pruning
+// routine.
+func WithDisabledGC() Option {
+	return func(p *Params) {
+		p.gcCycle = time.Duration(0)
+	}
+}
```

Comment:
I think we can get rid of this option in favor of WithGCCycle(0) to disable service.

Reply:
We launch it disabled for archival pruner -- eventually archival pruner might do *some* sort of pruning, but for now, we disable the service so it doesn't run since there's nothing to prune rn.

Reply:
Disabling service is already controlled by `EnableService` bool in nodebuilder config. There is no need to have this setting on service level. My personal take is that disabled services should not be a valid state.  Additionally, the lack of usage of  `WithDisabledGC()` in code suggests it is not needed and can be removed.

Reply:
https://github.com/celestiaorg/celestia-node/pull/3150/commits/d69a5cbbfa4701bd849d2782392c01bab63b4614